src/prefect/environments/execution/dask/k8s.py

```diff
@@ -265,6 +274,10 @@ def run_flow(self) -> None:
             )
             raise exc
 
+        # Call on_exit callback if specified
+        if self.on_exit:
+            self.on_exit()
```

Comment:
I wonder if users will expect that this is called _no matter what_ (meaning, prior to the exception raise above).  What do you think?

Reply:
I was considering that as well and I meant to add a comment to discuss. I think it should happen regardless. Should I move it into a `finally`?

Reply:
Yup, agreed - a `finally` sounds great.